parser/calcite/calcite_parser_test.go

```diff
@@ -54,4 +58,12 @@ func TestCalciteParser(t *testing.T) {
 	i, e = Parse("SELECT * FROM t1 t2, t3 t4 TO TRAIN DNNClassifier") // t2 and t4 are aliases.
 	a.Equal(27, i)
 	a.NoError(e)
+
+	i, e = Parse("SELECT * FROM (SELECT * FROM t1)")
```

Comment:
Please comment why this differs from tidbparser, can help future work, Thanks~

Reply:
This is the SQL standard. I commented in tidb_parser_test.go to note that TiDB and MySQL doesn't conform with SQL standard.